### ISSUE TITLE
extract path value from dir() object / document '!' syntax

### ISSUE THREAD

**Issue by dimpase** [gap-system/gap#5221]
While I know that to extract the path from a dir() object, say, `d`, one merely has to do `d![1]`
```
gap> d:=DirectoryCurrent();
dir("./")
gap> d![1];
"./"
```
this is not documented, i.e. by right users should not use this in their code. Are there plans to have a dedicated function for this?

Or, just document the `!` thing and live with it...

**Comment by fingolfin** [comment]
Use `Filename(d, "")`

**Comment by dimpase** [comment]
OK - except that it's not a filename that it returns, but a path. Hard to guess that it does  what it does.

**Comment by fingolfin** [comment]
I agree this is not convenient or nice, the whole `Filename` and `Directory` system is rather limited and incomplete. But I am not aware of any plans to improve this. Thoughtful contributions certainly would be welcome, though.

**Comment by fingolfin** [comment]
Oh and `Filename` always returns a "path", a better name for it might be `PathToFile` or something like that. But again, that's what it is, and while one can add new "better" functions, we are stuck with `Filename` as it is for now, to maintain compatibility with packages

**Comment by dimpase** [comment]
> Oh and `Filename` always returns a "path", a better name for it might be `PathToFile` or something like that. But again, that's what it is, and while one can add new "better" functions, we are stuck with `Filename` as it is for now, to maintain compatibility with packages

can we just have a copy of `Filename` called `Path`, or `DirectoryPath`, or..., and deprecate `Filename` ?
